arcgis-ios-sdk-samples/Scenes/Orbit camera around object/OrbitCameraAroundObjectViewController.swift

```diff
@@ -0,0 +1,185 @@
+        }
+    }
+    
+    func centerViewBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        guard let cameraController = sceneView.cameraController as? AGSOrbitGeoElementCameraController else { return }
+        
+        cameraController.isCameraDistanceInteractive = true
+        cameraController.isAutoPitchEnabled = false
+        
+        cameraController.targetOffsetX = 0
+        cameraController.targetOffsetY = 0
+        cameraController.targetOffsetZ = 0
+        
+        cameraController.cameraHeadingOffset = 0
+        
+        cameraController.minCameraPitchOffset = 10
+        cameraController.maxCameraPitchOffset = 100
+        cameraController.cameraPitchOffset = 45
+        
+        cameraController.minCameraDistance = 10
+        cameraController.cameraDistance = 50
+    }
+    
+    func cockpitViewBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        guard let cameraController = sceneView.cameraController as? AGSOrbitGeoElementCameraController else { return }
+        
+        cameraController.isCameraDistanceInteractive = false
+        cameraController.minCameraDistance = 0.1
+        // Unlock the camera pitch for the rotation animation.
+        cameraController.minCameraPitchOffset = -180
+        cameraController.maxCameraPitchOffset = 180
+        
+        // Animate the camera target to the cockpit.
+        cameraController.setTargetOffsetX(0, targetOffsetY: -2, targetOffsetZ: 1.1, duration: 1)
+        
+        // If the camera is already tracking the plane's pitch, subtract it from
+        // the delta angle for the animation.
+        let pitchDelta = cameraController.isAutoPitchEnabled ? 0 : 90 - cameraController.cameraPitchOffset + ((planeGraphic.attributes["PITCH"] as? NSNumber)?.doubleValue ?? 0)
+        // Animate the camera so that it is at the target (cockpit), facing
+        // forward (0 deg heading), and aligned with the horizon (90 deg pitch).
+        cameraController.moveCamera(
+            withDistanceDelta: 0.1 - cameraController.cameraDistance,
+            headingDelta: -cameraController.cameraHeadingOffset,
+            pitchDelta: pitchDelta,
+            duration: 1
+        ) { _ in
+            // When the animation finishes, lock the camera pitch.
+            cameraController.minCameraPitchOffset = 90
+            cameraController.maxCameraPitchOffset = 90
+            cameraController.isAutoPitchEnabled = true
+        }
+    }
+    
+    // MARK: UIViewController
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if let controller = segue.destination as? OrbitCameraSettingsViewController {
+            controller.orbitGeoElementCameraController = sceneView.cameraController as? AGSOrbitGeoElementCameraController
+            controller.planeGraphic = planeGraphic
+            controller.presentationController?.delegate = self
+            controller.preferredContentSize = CGSize(width: 300, height: 210)
```

Comment:
Instead of using a fixed size for the popover, consider having the settings controller size itself based on its contents. Search for `tableViewContentSizeObservation` for examples of how to do that.